### PR TITLE
Ronaru/tenant

### PR DESCRIPTION
**Pull request by rohitnarula7176**
*Updated code to retrieve tenantId from idToken

### CODE REVIEW

File: MSAL/src/MSALResult.m [hunk 1]
```diff
@@ -81,7 +82,7 @@ + (MSALResult *)resultWithTokenResult:(MSIDTokenResult *)tokenResult
     }
 
     MSIDAccount *resultAccount = tokenResult.account;
-    NSString *tenantId = [tokenResult.authority.url msidTenant];
+    NSString *tenantId = tokenResult.tokenResponse.idTokenObj.realm;
```
Comment (oldalton) [inline MSAL/src/MSALResult.m]: Will token response always be present? (e.g. in case of silent token lookups)

Comment (rohitnarula7176) [reply MSAL/src/MSALResult.m]: Done. Changed code to extract claims from rawIdToken.

File: MSAL/src/MSALResult.m [hunk 2]
```diff
@@ -81,7 +82,17 @@ + (MSALResult *)resultWithTokenResult:(MSIDTokenResult *)tokenResult
     }
 
     MSIDAccount *resultAccount = tokenResult.account;
-    NSString *tenantId = [tokenResult.authority.url msidTenant];
+    NSError *claimsError = nil;
```
Comment (oldalton) [inline MSAL/src/MSALResult.m]: Would be nice to add a unit test for this

Comment (rohitnarula7176) [reply MSAL/src/MSALResult.m]: Done. Added a new test class to test MSALResult.

**Pull request merged**